presto-main/src/main/java/io/prestosql/sql/planner/ExpressionInterpreter.java

```diff
@@ -900,7 +907,12 @@ protected Object visitFunctionCall(FunctionCall node, Object context)
 
             // do not optimize non-deterministic functions
             if (optimize && (!function.isDeterministic() || hasUnresolvedValue(argumentValues) || node.getName().equals(QualifiedName.of("fail")))) {
-                return new FunctionCall(node.getName(), node.getWindow(), node.isDistinct(), toExpressions(argumentValues, argumentTypes));
+                verify(!node.isDistinct(), "window does not support distinct");
```

Comment:
Check filter and orderBy here too? Maybe have a copy constructor for the builder?